server/src/pages/shot/text-tool.js

```diff
@@ -0,0 +1,205 @@
+    };
+  }
+
+  componentDidMount() {
+    this.textInput.current.focus();
+
+    if (this.props.toolbarOverrideCallback) {
+      this.props.toolbarOverrideCallback();
+    }
+  }
+
+  componentDidUpdate(oldProps, oldState) {
+    if (oldState.textSize !== this.state.textSize) {
+       this.props.toolbarOverrideCallback();
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.props.toolbarOverrideCallback) {
+      this.props.toolbarOverrideCallback();
+    }
+  }
+
+  render() {
+    const textToolBox = this.renderTextBox();
+    return <div
+      className={`text-tool-container centered`}
+      onDragOver={this.onDragOver.bind(this)}
+      onDrop={this.onDrop.bind(this)}
+      style={{height: this.props.baseCanvas.style.height, width: this.props.baseCanvas.style.width}}>
+      {textToolBox}
+    </div>;
+  }
+
+  renderTextBox() {
+    const dragDivTopPx = `${this.state.top - DRAG_PADDING}px`;
+    const dragDivLeftPx = `${this.state.left - DRAG_PADDING}px`;
+    const dragDivPadding = `${DRAG_PADDING}px`;
+
+    const dragDivStyles = {
+      position: "absolute",
+      top: dragDivTopPx,
+      left: dragDivLeftPx,
+      padding: dragDivPadding,
+      cursor: "move"
+    };
+
+    return [
+      <div draggable="true" key="drag" onDragStart={this.onDragStart.bind(this)} style={dragDivStyles}>
+        <div id="text-input" ref={this.textInput} contentEditable="true" key="text"
+           className={`${this.state.textSize} ${this.state.colorName} text`} >
+        </div>
+      </div>
+    ];
+  }
+
+  renderToolbar() {
+    return <div className="editor-header default-color-scheme"><div className="annotation-tools">
+      <Localized id="annotationTextSize">
+        <select className={`text-select`} title="Text Size" onChange={this.onChangeTextSize.bind(this)} value={this.state.textSize}>
+          <Localized id="textSizeSmall"><option value="small-text">Small</option></Localized>
+          <Localized id="textSizeMedium"><option value="medium-text">Medium</option></Localized>
+          <Localized id="textSizeLarge"><option value="large-text">Large</option></Localized>
+        </select>
+      </Localized>
+      <ColorPicker activeTool={this.state.tool}
+        setColorCallback={this.setColor.bind(this)}
+        color={this.state.color} />
+      <span className="annotation-divider"></span>
+      <Localized id="annotationTextConfirmButton">
+        <button className={`button transparent confirm-text`} id="confirm-text" onClick={this.onClickConfirm.bind(this)} title="Confirm Text">Confirm</button>
+      </Localized>
+      <Localized id="annotationTextCancelButton">
+        <button className={`button transparent cancel-text`} id="cancel-text" onClick={this.onClickCancel.bind(this)} title="Cancel Text">Cancel</button>
+      </Localized>
+    </div></div>;
+  }
+
+  setColor(color, colorName) {
+    this.setState({color, colorName});
+    this.textInput.current.focus();
+  }
+
+  onClickConfirm(e) {
```

Comment:
If the user did not enter any text, we probably shouldn't save anything onto the canvas.

Reply:
Done